google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/SubscriberSnippets.java

```diff
@@ -86,8 +87,11 @@ private void createSubscriber() throws Exception {
     MessageReceiver receiver = new MessageReceiver() {
           @Override
           public void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
-            // handle incoming message, then ack or nack the received message
-            // ...
+            // handle incoming message, then ack/nack the received message
+            System.out.println("Id : " + message.getMessageId());
+            // decode from base64 to view original data
+            System.out.println("Data : " +
+                new String(DatatypeConverter.parseBase64Binary(message.getData().toStringUtf8())));
```

Comment:
Why do we need to decode base64? Is the original data base64 encoded? Pubsub works OK with binary data, doesn't it?